src/Compilers/CSharp/Test/Symbol/Symbols/Source/FileModifierTests.cs

```diff
@@ -2497,5 +2946,185 @@ namespace System
         Assert.Equal("System.Void@<tree 0>", typeInfo.Type!.ToDisplayString(SymbolDisplayFormat.TestFormat.WithCompilerInternalOptions(SymbolDisplayCompilerInternalOptions.IncludeContainingFileForFileTypes)));
     }
 
-    // PROTOTYPE(ft): public API (INamedTypeSymbol.IsFile?)
+    [Fact]
+    public void GetTypeByMetadataName_01()
+    {
+        var source1 = """
+            file class C { }
+            """;
+
+        // from source
+        var comp = CreateCompilation(source1);
+        comp.VerifyDiagnostics();
+        var sourceMember = comp.GetMember<NamedTypeSymbol>("C");
+        Assert.Equal("<>F0__C", sourceMember.MetadataName);
+
+        var sourceType = comp.GetTypeByMetadataName("<>F0__C");
+        Assert.Equal(sourceMember, sourceType);
+
+        Assert.Null(comp.GetTypeByMetadataName("<>F0__D"));
+        Assert.Null(comp.GetTypeByMetadataName("<>F1__C"));
+        Assert.Null(comp.GetTypeByMetadataName("F0__C"));
+        Assert.Null(comp.GetTypeByMetadataName("<file>F0__C"));
```

Comment:
Conversely, all the positive `GetTypeByMetadataName` tests so far seem to have an empty file name. Let's add one where the syntax tree is named.